tests/endpoints/bookstore_grpc/server/server.go

```diff
@@ -266,6 +267,22 @@ func (s *BookstoreServerV1Impl) CreateShelf(ctx context.Context, req *bspbv1.Cre
 		return nil, err
 	}
 
+	// Unmarshal, deepcopy and marshal, to verify the received binary `any`
+	if req.Shelf.Any != nil {
+		var book bspbv1.Book
+		err := ptypes.UnmarshalAny(req.Shelf.Any, &book)
+		if err != nil {
+			return nil, status.New(codes.Internal, "cannot unmarshal shelf.any").Err()
+		}
+
+		newBook := book
+
+		req.Shelf.Any, err = ptypes.MarshalAny(&newBook)
```

Comment:
What is the purpose of marshaling into Any again?  I think we don't need it

Reply:
The normal test procedure I think would be 
- sending a json request, getting/unmarshalling received any request and verify it
- sending an any response, getting received any response and verify it.

I choose to verify only on client side with the echoed request/response, since here we are using a real backend and I don't want add too much test checking logic there. 

We can believe json-any transcoding is correct by:
- the echoed json is correct
- the backend can successfully unmarshal the `Any`, which is more convincing than sending back binary `Any` directly.